### PR TITLE
Adding support for IOS recordError function

### PR DESCRIPTION
**Pull request by nbdamian**
In reference to issue: #11 
I have added support for recordError on IOS, and fixed some spelling mistakes, etc.
Note this is untested as I don't have an IOS device with me right now. Please test it before merging.

### CODE REVIEW

File: src/ios/FabricPlugin.m [hunk 1]
```diff
@@ -246,6 +247,11 @@ - (void)sendCrash:(CDVInvokedUrlCommand*)command
     [[Crashlytics sharedInstance] crash];
 }
 
+- (void)recordError:(CDVInvokedUrlCommand*)command
+{
+    [[Crashlytics sharedInstance] recordError];
```
Comment (Justin-Credible) [inline src/ios/FabricPlugin.m]: This is a compilation error; the `recordError` selector takes an `NSError` object.

Also, the `message` parameter is not being passed here.

File: www/FabricPlugin.Answers.js [hunk 2]
```diff
@@ -1,87 +1,62 @@
-'use strict';
```
Comment (Justin-Credible) [inline www/FabricPlugin.Answers.js]: This file should be JavaScript, but it now contains es6 code. It looks like the something went wrong while transpiling the code. This breaks the entire plugin on iOS.

File: www/FabricPlugin.Crashlytics.js [hunk 3]
```diff
@@ -1,67 +1,51 @@
-'use strict';
```
Comment (Justin-Credible) [inline www/FabricPlugin.Crashlytics.js]: This file should be JavaScript, but it now contains es6 code. It looks like the something went wrong while transpiling the code. This breaks the entire plugin on iOS.

File: www/FabricPlugin.js [hunk 4]
```diff
@@ -1,29 +1,16 @@
-'use strict';
+const exec = require('cordova/exec');
```
Comment (Justin-Credible) [inline www/FabricPlugin.js]: This file should be JavaScript, but it now contains es6 code. It looks like the something went wrong while transpiling the code. This breaks the entire plugin on iOS.

**Comment by Justin-Credible** [comment]
Also, as a side note, a lot of whitespace was changed in this PR. This makes the diff harder to review. Please consider reverting your whitespace changes.

**Comment by nbdamian** [comment]
Regarding the whitespace, I believe it is atom.io adding it, I will check to see if I can disable it.
I'm not sure why the transpile is not working, All I am running is "npm run-script transpile" (Is this not correct?).
Unless this is an error with Ubuntu.

**Comment by Justin-Credible** [comment]
To transpile, this should work:
````
 $ npm install
 $ npm run-script transpile

> cordova-fabric-plugin@1.0.5 transpile /tmp/FabricPlugin-nbdamian
> babel www/*.es6 --out-dir .

www/FabricPlugin.Answers.es6 -> www/FabricPlugin.Answers.js
www/FabricPlugin.Crashlytics.es6 -> www/FabricPlugin.Crashlytics.js
www/FabricPlugin.es6 -> www/FabricPlugin.js
````

The output should look similar to the original JavaScript files.

**Comment by nbdamian** [comment]
OK, the transpile has been fixed. I am currently looking into the NSError.

**Comment by sarriaroman** [comment]
@Justin-Credible are we ok to accept the merge?

**Comment by nbdamian** [comment]
@sarriaroman I would say wait until I have fixed the compilation error.

**Comment by nokrasnov** [comment]
@nbdamian could you fix error, please? I (and probably many other people) would certainly appreciate it.

**Comment by nbdamian** [comment]
Sorry, in the last 5 months, our 2 iMacs and 1 macbook pro have broken (graphic card issues). I'll push the changes I have made that should allow it build.
Ill push it first thing in the morning.

Sent using Boxer

On May 4, 2016 8:47 PM, Nikita Krasnov <notifications@github.com> wrote:

@nbdamian<https://github.com/nbdamian> could you fix error, please? I (and probably many other people) would certainly appreciate it.

-

**Comment by nbdamian** [comment]
Ok, I have updated the request, could someone test to see if it is letting you build now.
Thanks...

File: src/ios/FabricPlugin.m [hunk 5]
```diff
@@ -246,6 +247,11 @@ - (void)sendCrash:(CDVInvokedUrlCommand*)command
     [[Crashlytics sharedInstance] crash];
 }
 
+- (void)recordError:(CDVInvokedUrlCommand*)command
+{
+    [[Crashlytics sharedInstance] recordError:error]
```
Comment (Justin-Credible) [inline src/ios/FabricPlugin.m]: This method requires an `NSError` object. Also, the error variable does not exist so this is a compilation error.

File: www/FabricPlugin.Crashlytics.es6 [hunk 6]
```diff
@@ -19,6 +19,11 @@ class FabricCrashlytics {
         ]);
     }
 
+    recordError() {
+        window.fabric.core.execPlugin('recordError', [
```
Comment (Justin-Credible) [inline www/FabricPlugin.Crashlytics.es6]: The document says `recordError` accepts a message parameter, but it is not passed here.

**Comment by Justin-Credible** [comment]
Have you tested this code? I don't think it will compile. See the comments above.

Also, review your diff and revert all whitespace changes please.

**Comment by nbdamian** [comment]
As I said, I am unable to test it as I don't have a Apple computer any more.

**Comment by Justin-Credible** [comment]
We're definitely willing to accept a PR for this issue, but the PR should compile and the submitter should  test to ensure it works as expected. This will involve an acceptance test that shows, at a minimum, the value passed into the `recordError` method showing up in the Fabric console.

Once we have a PR that meets this criteria we can run the acceptance test on our end to verify and then we'll accept the PR.

I'm closing this for now. Feel free to open a new PR if you're able to address these concerns. Thanks!

**Pull request closed without merging**